javascript/javascript1/week3/warmuptask.js

```diff
@@ -0,0 +1,69 @@
+//Item array removal
+const names = [
+  "Peter",
+  "Ahmad",
+  "Yana",
+  "kristina",
+  "Rasmus",
+  "Samuel",
+  "katrine",
+  "Tala",
+];
+const nameToRemove = "Ahmad";
+function removeTheName(name) {
+  return name !== nameToRemove;
+}
+const newArray = names.filter(removeTheName);
+console.log(newArray);
```

Comment:
Here are a couple of other solutions

1) Using the [indexOf](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/indexOf) method
```
const nameToRemove = "Ahmad";

const indexOfNameToRemove = names.indexOf(nameToRemove);

if (indexOfNameToRemove >= 0) {
  names.splice(indexOfNameToRemove, 1);
} else {
  console.log("Informative error message");
}

console.log(names);
```

2) Looping through the array and comparing each name, then using `splice` method
```
const nameToRemove = "Ahmad";

for (let i = 0; i < names.length; i++) {
  if (names[i] === nameToRemove) {
    names.splice(i, 1);
    break; // prevents looping more, as soon as it finds the matching name
  }
}
console.log(names);
```

Reply:
Thank you for providing additional solutions.